src/lib.rs

```diff
@@ -0,0 +1,197 @@
+	>;
+
+	#[pallet::event]
+	// #[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		SomethingStored(u32, T::AccountId),
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		NoneValue,
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_initialize(n: T::BlockNumber) -> Weight {
+			Self::do_execute_subscriptions(n)
+		}
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::weight(1)]
+		pub fn create_subsciption_plan(
+			_origin: OriginFor<T>,
+			_plan: Subscription<T::BlockNumber, BalanceOf<T>, T::AccountId>,
+		) -> DispatchResult {
+			// maybe make a new function for Subscription, so `plan` is already valid 100%
+			//
+			//
+			// todo:
+			// signed ?
+			// valid plan ?
+			//
+			// -> push the plan to the storage
+			Ok(())
+		}
+
+		#[pallet::weight(1)]
+		pub fn delete_subsciption_plan(_origin: OriginFor<T>, _plan_id: Nonce) -> DispatchResult {
+			// better ? 5 parameters but we can check, better option if no plan.new() function
+			//
+			//
+			// todo:
+			// signed ?
+			// plan exist ?
+			//
+			// -> remove the plan to the storage
+			Ok(())
+		}
+
+		#[pallet::weight(1)]
+		pub fn subscribe_to_plan(_origin: OriginFor<T>, _plan_id: Nonce) -> DispatchResult {
+			// todo:
+			// signed ?
+			// plan exist ?
+			//
+			// -> create entry in storage
+			Ok(())
+		}
+
+		#[pallet::weight(1)]
+		pub fn subscribe(
+			_origin: OriginFor<T>,
+			_plan: Subscription<T::BlockNumber, BalanceOf<T>, T::AccountId>,
+		) -> DispatchResult {
+			// todo:
+			// signed ?
+			// valid subscription ?
+			//
+			// -> create entry in storage
+			Ok(())
+		}
+
+		#[pallet::weight(1)]
+		pub fn unsubscribe(_origin: OriginFor<T>, _other: T::AccountId) -> DispatchResult {
+			// signed by the subscriber ?
+			// yes -> subscriber is subscribed to other ?
+			//        delete from storage
+			// no -> signed by beneficiary ?
+			//       other is subscribed to beneficiary ?
+			//       delete from storage
+			Ok(())
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		pub fn do_execute_subscriptions(n: T::BlockNumber) -> Weight {
+			let mut itterations = 0;
+
+			let subs = unwrap_or_return!(Subscriptions::<T>::get(n), 0);
+			for (mut sub, account) in subs {
+				if let Some(val) = sub.remaining_payments {
+					if val == 0 {
+						continue
+					}
+				}
+				itterations += 1;
+				if T::Currency::transfer(
+					&account,
+					&sub.beneficiary,
+					sub.amount,
+					ExistenceRequirement::AllowDeath,
+				)
+				.is_err()
+				{
+					continue
+				}
+				sub.remaining_payments = sub.remaining_payments.map(|amount| amount - 1);
+				let new_block = n + sub.frequency;
+                let new_subs = Subscriptions::<T>::take(new_block).map(|mut subs| {
+                    subs.push((sub.clone(), account.clone()));
+                    subs
+                }).unwrap_or_else(|| vec!((sub, account)));
+                Subscriptions::<T>::insert(new_block, new_subs);
```

Comment:
Doing this you will do a read and a write for each subscription.
You should rather store the value you updated in a on-memory cache, update this cache as many time as you want and and only write them back to storage at the end of the whole loop